src/releasers/index.ts

```diff
@@ -27,30 +27,50 @@ import {TerraformModule} from './terraform-module';
 import {Rust} from './rust';
 import {OCaml} from './ocaml';
 
-// TODO: add any new releasers you create to this list:
-export function getReleasers(): {[key: string]: typeof ReleasePR} {
-  const releasers = {
-    go: GoYoshi, // TODO(codyoss): refactor this into a more generic go strategy.
-    'go-yoshi': GoYoshi,
-    'java-bom': JavaBom,
-    'java-yoshi': JavaYoshi,
-    node: Node,
-    'php-yoshi': PHPYoshi,
-    python: Python,
-    'ruby-yoshi': RubyYoshi,
-    ruby: Ruby,
-    simple: Simple,
-    'terraform-module': TerraformModule,
-    rust: Rust,
-    ocaml: OCaml,
-  };
+// add any new releasers you create to this type as well as the `releasers`
+// object below.
+export type ReleaseType =
+  | 'base' // not exposed to end users
+  | 'go'
+  | 'go-yoshi'
+  | 'java-bom'
+  | 'java-yoshi'
+  | 'node'
+  | 'php-yoshi'
+  | 'python'
+  | 'ruby-yoshi'
+  | 'ruby'
+  | 'simple'
+  | 'terraform-module'
+  | 'rust'
+  | 'ocaml';
+
+type Releasers = Partial<Record<ReleaseType, typeof ReleasePR>>;
+
+const releasers: Releasers = {
+  go: GoYoshi, // TODO(codyoss): refactor this into a more generic go strategy.
+  'go-yoshi': GoYoshi,
+  'java-bom': JavaBom,
+  'java-yoshi': JavaYoshi,
+  node: Node,
+  'php-yoshi': PHPYoshi,
+  python: Python,
+  'ruby-yoshi': RubyYoshi,
+  ruby: Ruby,
+  simple: Simple,
+  'terraform-module': TerraformModule,
+  rust: Rust,
+  ocaml: OCaml,
+};
+
+export function getReleasers(): Releasers {
   return releasers;
 }
 
-export function getReleaserNames(): string[] {
```

Comment:
Note that this is a breaking change (the GitHub bot references this one)

Reply:
oh, good point. Should I leave it in (calling `getReleaserTypes()` under the hood)?

Reply:
if so is it ok if the return type is `ReleaseType[]`?

Reply:
I think it should be. `ReleaseType` is a limited set of strings and the typing is only for compile time.